lib/Dialect/TritonNvidiaGPU/Transforms/FenceInsertion.cpp

```diff
@@ -32,16 +33,22 @@ struct FenceInsertionPass
   FenceInsertionPass(int computeCapability) {
     this->computeCapability = computeCapability;
   }
-  // TODO: support more patterns to insert fences
-  // only support insertion between convert layout ops and dot ops to protect
-  // flashattention
+  // TODO: support more general patterns to insert fences.
+  // generic + fence + async
+  // eg. any op(generic) to shared in use-def chain which refers by async proxy
   void runOnOperation() override {
     // Only insert fences for compute capability 9.0
     if (computeCapability < 90)
       return;
+    // ENABLE_TMA and ENABLE_MMA_V3
+    if (!::triton::tools::getBoolEnv("ENABLE_TMA") or
+        !::triton::tools::getBoolEnv("ENABLE_MMA_V3"))
```

Comment:
In a sense that the fence is between stmatrix and tmastg, why do we care ENABLE_MMA_V3?